.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright 2024 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+#
+
+name: Build and Test
+
+on:
+  pull_request:
+
+jobs:
+  build-and-test:
+    name: " "
+    uses: adobe/aepsdk-commons/.github/workflows/ios-build-and-test.yml@gha-ios-5.0.0
+    with:
+      ios-device-names: '["iPhone 15"]'
+      ios-versions: '["18.1"]'
+      tvos-device-names: '["Apple TV"]'
+      tvos-versions: '["18.1"]'
+      run-test-ios-unit: true
+      run-test-ios-functional: true
+      run-test-ios-integration: true
+      run-test-tvos-unit: true
+      run-test-tvos-functional: true
+      run-build-xcframework-and-app: true
```

Comment:
Is it possible to only run some of these tests on specific branches? For example, we run the integration tests and build the xcframework and app just on the staging and main branches to reduce the time it takes to build the dev and pr branches. Depending on how long it takes to run these jobs it may not be necessary to exclude them, however.

Reply:
ok, then let's see how this plays out.